### PR TITLE
Task 4748 - Refactor for Menus

### PR DESCRIPTION
**Pull request by Sarin2023**
1) Popover menu list of options
2) Method to get all options (".v-list-item") as ListItem collection

### CODE REVIEW

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/MenusTests.java [hunk 1]
```diff
@@ -210,7 +210,8 @@ public void menuAsPopoverMenuTests() {
         menuAsPopoverButton.click();
         activeMenu.is().displayed();
         activeMenu.has().numberOfOptions(3);
-        activeMenu.has().optionsTitles(OPTIONS_TITLES_NAMES);
+        //activeMenu.has().optionsTitles(OPTIONS_TITLES_NAMES);
```
Comment (pnatashap) [inline jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/MenusTests.java:213]: Comments are not allowed, keep only valid code

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java [hunk 2]
```diff
@@ -20,6 +22,16 @@ public String hasPosition() {
         return String.format("top: %s; left: %s", core().getCssValue("top"), core().getCssValue("left"));
     }
 
+    @JDIAction("'{name}' has position from top")
+    public String hasTopPosition() {
```
Comment (pnatashap) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java:26]: Wrong name.
If method has name has* that means that the result type is boolean, the same as is* 

Also it is a very bad idea to combine the string "top: ", typically we need a real number to use in autotests

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/MenusTests.java (hunk 1 above)
Comment (Sarin2023) [reply jdi-light-vuetify-tests/src/test/java/io/github/epam/vuetify/tests/complex/MenusTests.java]: Corrected

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java (hunk 2 above)
Comment (Sarin2023) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java]: Corrected

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/MenuAssert.java [hunk 3]
```diff
@@ -37,6 +49,12 @@ public MenuAssert optionsTitles(List<String> optionsTitles) {
         return this;
     }
 
+    @JDIAction("Assert that '{name}' has expected options")
+    public MenuAssert optionsList(List<String> optionsTitles) {
```
Comment (pnatashap) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/MenuAssert.java:53]: It is very difficult to use if we have optionsList in validation and getOptions in element. They must have similar names

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java [hunk 4]
```diff
@@ -20,6 +22,16 @@ public String hasPosition() {
         return String.format("top: %s; left: %s", core().getCssValue("top"), core().getCssValue("left"));
     }
 
+    @JDIAction("'{name}' has position from top")
+    public int getTopPosition() {
```
Comment (pnatashap) [inline jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java:26]: topPosition() is good name, we have a minimum methods with get*

**Review by pnatashap — CHANGES_REQUESTED** [review]

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java (hunk 4 above)
Comment (Sarin2023) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/elements/complex/Menu.java]: Modified the names.

File: jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/MenuAssert.java (hunk 3 above)
Comment (Sarin2023) [reply jdi-light-vuetify/src/main/java/com/epam/jdi/light/vuetify/asserts/MenuAssert.java:53]: Modified the names.

**Pull request merged**